students/ethan_nguyen/lesson06/mailroom4.py

```diff
@@ -0,0 +1,179 @@
+
+def update_amount(name, amount):
+    """function to update amount given a name"""
+    donor_db[name].append(amount)
+
+def add_new_donor(name, amount):
+    """function to add a new donor"""
+    donor_db[name] =[amount]
+
+def sort_key(donor):
+    return sum(donor[1])
+
+def get_report():
+    sort_Donor = sorted(donor_db.items(), key=sort_key, reverse = True)
+    rows = list()
+    for d in sort_Donor:
+        rows.append("{:<25} ${:>12,.2f}{:>22}{:<10}${:>12,.2f}".format(d[0], sum(d[1]), len(d[1]), '',sum(d[1])/len(d[1])))
+    return rows
+
+def display_report():
+    for r in get_report():
+        print(r)
+
+def create_report():
+
+    space=''
+    print(f'Donor Name {space:<15}| Total Given {space:<10}| Num Gifts {space:<9}| Average Gift {space:<19}')
+    print("-"*85)
+    #sort_Donor = sorted(donor_db.items(), key=lambda x: x[1], reverse = True)
+    display_report()
+
+def get_letter_text(name, amount):
+    """Get letter text for file content"""
+    return f'Dear {name}, \n Thank you for your very kind donation of ${float(sum(amount)):,.2f} \n \
+        It will be put to very good use. \n Sincerely, \n -UW'
+
+def create_letter(path, letter_name, letter):
+    """Create letter"""
+    with open(f'{path}/{letter_name}.txt', 'w') as f:
+        f.write(letter) 
+
+def send_letters():
+
+    path = os.getcwd()
+    for k,v in donor_db.items():
+        create_letter(path, k, get_letter_text(k, v))
+    
+    print("Done!")
+
+def test_get_report():
+    expected = "{:<25} ${:>12,.2f}{:>22}{:<10}${:>12,.2f}".format("Paul Allen", 708.42, 3, '',236.14)
+
+    result = get_report()
+    assert expected in result
+
+def test_get_report_count():
+
+    result = get_report()        
+    assert len(result) == 5
+
+def test_get_letter_text():
+    expected = "Dear World, \n Thank you for your very kind donation of $300.00 \n \
+        It will be put to very good use. \n Sincerely, \n -UW"
+    assert get_letter_text("World", [100, 200]) == expected
+
+def test_create_letter():
+    expected = os.path.join(os.getcwd(), "World.txt")
+    create_letter(os.getcwd(), "World", "Dear World, \n Thank you for your very kind donation of $300.00 \n \
+        It will be put to very good use. \n Sincerely, \n -UW")
+    assert os.path.isfile(expected)
+
+def test_create_thank_you_note():
+    expected = 'Dear Ethan, \n Thank you for your very kind donation of $10,000.00 \n \
+        It will be put to very good use. \n Sincerely, \n -UW'
+    
+    assert create_thank_you_note("Ethan", 10000) == expected
+
+def test_add_new_donor():
+
+    add_new_donor('Ethan', 10000)
+    expected_key, expected_value =  "Ethan", [10000]
+
+    assert expected_key in donor_db and expected_value == donor_db[expected_key]
+
+def test_update_amount():
+
+    update_amount("Ethan", 100)
+    expected_key, expected_value =  "Ethan", [10000, 100]
+    assert expected_key in donor_db and expected_value == donor_db[expected_key]
```

Comment:
your actual tests look good - however all of your tests should be in a separate file called `test_mailroom.py`